### DISCUSSION TITLE
Is ArcWelderPlugin compatible to OctoPrint Integration Plugin in Cura

### DISCUSSION THREAD

**Discussion by NilsRo** [FormerLurker/ArcWelderPlugin#221] [Q&A]
Hi,

as Curas OctoPrint Integration Plugin made by @fieldofview uses an API to upload and directly starts a print I ask me if the plugin supports this. There is also an ArcWelder Plugin in Cura but I am thinking of using it in a central position. Perhaps you can answer that before I break something or it is not used in the background.

Many thanks,
Nils

**Comment by fieldOfView** [comment]
As the author of both the Cura OctoPrint Connection plugin and the ArcWelder plugin for Cura, , I recommend running ArcWelder from Cura. That way you benefit from the (probably) much better performance of the computer you use to slice with to process the gcode file. 

If you use the ArcWelder plugin on the side of OctoPrint, automatic printing after upload from Cura probably does not work well because a different file than the one that gets uploaded must be printed, and it takes a while before that file is available and ready. If you do the processing from inside Cura, then that is not an issue.

I would not really know how the OctoPrint Connection plugin in Cura would be able to see if the ArcWelder plugin on the OctoPrint side is done processing a gcode file so it can tell OctoPrint to start that printing that ArcWelded gcode file.
